source/apiVolontaria/volunteer/serializers.py

```diff
@@ -414,11 +418,48 @@ class ParticipationBasicSerializer(serializers.ModelSerializer):
 
     # Explicitly declare the BooleanField to make it "required"
     standby = serializers.BooleanField()
+    invit_icalendar = serializers.BooleanField(
+        default=False,
+        write_only=True,
+    )
     user = UserBasicSerializer(
         read_only=True,
         default=serializers.CurrentUserDefault(),
     )
 
+    def create(self, validated_data):
+        invitation_ics = validated_data.pop('invit_icalendar')
+        participation = models.Participation.objects.create(**validated_data)
+
+        if invitation_ics and settings.CONSTANT['EMAIL_SERVICE'] is True:
+            if (os.path.exists(settings.MEDIA_ROOT) == False):
+                os.makedirs(settings.MEDIA_ROOT)
+
+            file_event_path = settings.MEDIA_ROOT + '/' + '%.0f' % timezone.now().timestamp() + '.ics'
+
+            functions.generate_participation_ics(participation, file_event_path)
+            if os.path.isfile(file_event_path):
+                merge_data = {
+                    'DOWNLOAD_URL': file_event_path,
+                    'CSS_STYLE': render_to_string('css/calendar_invitation.css')
+                }
+
+                plain_msg = render_to_string("calendar_invitation.txt")
+                msg_html = render_to_string("calendar_invitation.html", merge_data)
+                service_send_mail([participation.user.email], _('invitation calandar'), plain_msg, msg_html)
+            else:
```

Comment:
The else is not on the good condition, did you try it ?
You will need to test it to be sure it work (add a test in test file).